### PR TITLE
Fix "registration is not supported" never displayed

### PR DESCRIPTION
**Pull request by uabarahona**
# Summary
This solves the problem where the "Registration is not supported" message were not being displayed, there were some exception due `RedirectToRegister()` was being called always so I added a test to check for exception in all states

- [x] Adding tests to confirm the issue
   - [x] Adding test to avoid exceptions during rendering
   - [x] Adding test to confirm appropriate Render code is being called
- [x] Fix the issue

Addresses #22912

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24862) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24862) before we can accept your contribution.<br/><br/>:x: barahonajm [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24862)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=24862) it.</sub>

File: src/Components/WebAssembly/WebAssembly.Authentication/test/RemoteAuthenticatorCoreTests.cs [hunk 1]
```diff
@@ -646,6 +719,13 @@ public TestRemoteAuthenticatorView()
                 };
             }
 
+            public TestRemoteAuthenticatorView(RemoteAuthenticationApplicationPathsOptions applicationPaths,
+                                               IJSRuntime jsRuntime = default)
```
Comment (javiercn) [inline src/Components/WebAssembly/WebAssembly.Authentication/test/RemoteAuthenticatorCoreTests.cs:723]:
```suggestion
            public TestRemoteAuthenticatorView(RemoteAuthenticationApplicationPathsOptions applicationPaths, IJSRuntime jsRuntime = default)
```
nit: We don't indent things this way

**Review by javiercn — COMMENTED** [review]
Looks great!

Thanks for the contribution!

File: src/Components/WebAssembly/WebAssembly.Authentication/test/RemoteAuthenticatorCoreTests.cs (hunk 1 above)
Comment (barahonajm) [reply src/Components/WebAssembly/WebAssembly.Authentication/test/RemoteAuthenticatorCoreTests.cs]: Thanks!

**Review by javiercn — APPROVED** [review]

**Comment by javiercn** [comment]
@barahonajm thanks for the contribution!

**Pull request merged**